src/main.rs

```diff
@@ -61,11 +70,25 @@ async fn main() -> Result<(), Box<dyn Error>> {
         let queue = queue.clone();
 
         async move {
-            Ok::<_, HyperError>(service::service_fn(move |_: Request<Body>| {
+            Ok::<_, HyperError>(service::service_fn(move |request: Request<Body>| {
                 let queue = queue.clone();
 
+                let mut shard = None;
+
+                if let Ok(url) = Url::parse(&request.uri().to_string()) {
+                    for (k, v) in url.query_pairs() {
+                        if k == "shard" {
+                            shard = v.parse::<u64>().ok();
+                        }
+                    }
+                }
+
+                if shard.is_none() && big_queue {
+                    info!("No shard id set, defaulting to 0. Will not bucket requests correctly!");
+                }
+
```

Comment:
The log could be a warn, apart from that this can be avoided if `big_queue` is false.

```suggestion
                let mut shard = None;

                if big_queue {
                    if let Ok(url) = Url::parse(&request.uri().to_string()) {
                        for (k, v) in url.query_pairs() {
                            if k == "shard" {
                                shard = v.parse::<u64>().ok();
                            }
                        }
                    }

                    if shard.is_none() {
                        warn!("No shard id set, defaulting to 0. Will not bucket requests correctly!");
                    }
                }
```